pkg/openstackconfiggenerator/configmap.go

```diff
@@ -383,7 +384,13 @@ func createRolesMap(
 
 	for _, osnet := range osNetList.Items {
 		for roleName, roleReservation := range osnet.Spec.RoleReservations {
-			if IsRoleIncluded(roleName, instance) {
+			var roleOveride ospdirectorv1beta1.TripleoRoleOverrideSpec
+			roleOverrideFound := false
+			if instance.Spec.TripleoRoleOverride != nil {
```

Comment:
should we add a check that this only works on OSP 17.1, or this also supported in 16.2?

Reply:
hrmm, well you could use it with 16.2 I guess, but not sure why you would want to

Reply:
so then, to prevent someone using it with 16.2 should we prevent that since the code if for both versions?

Reply:
Ok. Validating in a webhook would be very heavy since it would need to get the openstackcontrolplane to check the version. Could just return an error in the reconcile loop if it's not 17.1.

Reply:
yes, that works or we could do the same in the configgenerator webhook, or you say thats the heavy way?
```
if instance.Spec.TripleoRoleOverride != nil && ospVersion != shared.OSPVersion(shared.TemplateVersion17_1) {
		return &TripleoRoleOverrideVersionError{ospVersion}
	}
```